ci/Jenkinsfile.test-e2e.android

```
@@ -0,0 +1,277 @@
+            }
+            sh "mkdir -p pkg"
+            if (env.BUILD_SOURCE.startsWith('http')) {
+              sh """
+                set -euo pipefail
+                curl -L "${env.BUILD_SOURCE}" -o pkg/downloaded.apk
+              """
+            } else {
+              copyArtifacts(
+                projectName: env.BUILD_SOURCE,
+                filter: 'pkg/*.apk',
+                selector: lastWithArtifacts(),
+                target: 'pkg/'
+              )
+            }
+            def apkPath = utils.findFile('pkg/*.apk')
+            if (!apkPath) {
+              error("Unable to locate APK under pkg/. Ensure BUILD_SOURCE produces pkg/*.apk artifacts.")
+            }
+            env.APK_PATH = apkPath
+            echo "APK ready at ${env.APK_PATH}"
+          }
+        }
+      }
+    }
+
+    stage('Upload APK to BrowserStack') {
+      when {
+        expression { !env.BROWSERSTACK_APP_ID }
+      }
+      steps {
+        script {
+          withCredentials([
+            usernamePassword(
+              credentialsId: 'browserstack-status-desktop',
+              usernameVariable: 'BROWSERSTACK_USERNAME',
+              passwordVariable: 'BROWSERSTACK_ACCESS_KEY'
+            )
+          ]) {
+            def response = sh(
+              script: """
+                set -euo pipefail
+                APK_NAME=\$(basename "${env.APK_PATH}")
+                CUSTOM_ID=\$(printf '%s' "\${APK_NAME}" | tr -cs '[:alnum:]._-' '-' | cut -c1-80)
+                CUSTOM_ID="\${CUSTOM_ID}-${env.BUILD_NUMBER ?: System.currentTimeMillis()}"
+                curl -s -u "${BROWSERSTACK_USERNAME}:${BROWSERSTACK_ACCESS_KEY}" \\
+                  -X POST "https://api-cloud.browserstack.com/app-automate/upload" \\
+                  -F "file=@${env.APK_PATH}" \\
+                  -F "custom_id=\${CUSTOM_ID}" \\
+                | tee upload_response.json
+              """,
+              returnStdout: true
+            ).trim()
+            def result = readJSON text: response
+            def appUrl = result?.app_url
+            if (!appUrl) {
+              error("BrowserStack upload failed: ${response}")
+            }
+            env.BROWSERSTACK_APP_ID = appUrl
+            env.BROWSERSTACK_BUILD_NAME = result?.custom_id
+            env.BROWSERSTACK_BUILD_IDENTIFIER = env.BUILD_NUMBER
+            echo "BrowserStack app uploaded: ${env.BROWSERSTACK_APP_ID}"
+          }
+        }
+      }
+    }
+
+    stage('Run pytest suite') {
+      steps {
+        script {
+          dir('test/e2e_appium') {
+            if (!env.DEVICE_ID?.trim()) {
+              echo 'DEVICE_ID not set; using framework defaults.'
+            } else {
+              echo "Using DEVICE_ID: ${env.DEVICE_ID}"
+            }
+
+            def pytestCmd = "python -m pytest --env browserstack"
+            if (env.PYTEST_ARGS?.trim()) {
+              pytestCmd += " ${env.PYTEST_ARGS.trim()}"
+            }
```

I don't get it, why not just use it directly? What does the `if` provide?
```suggestion
            def pytestCmd = "-m pytest --env browserstack ${env.PYTEST_ARGS.trim()}"
```
Actually, the `def` is pointless, you should just call the command directly.